### PR TITLE
Writing unit tests for the files operate.js and calculate.js using Jest

### PR DESCRIPTION
**Pull request by Mutai-Gilbert**
In this pull request, we were able to:
- Write unit tests with React Testing Library.
- Tests for Components are added.

### CODE REVIEW

File: src/tests/Calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  });
+  test('when AC button is clicked', () => {
+    const output = calculate({
+      total: '8',
+      next: '5',
+      operation: '',
+    }, 'AC');
+    expect(output.total).toBe(null);
+  });
+  test('when % button is clicked', () => {
```
Comment (afanimartin) [inline src/tests/Calculate.test.js]: - Please test cases all modules. You are missing tests for the `.` and `+/-` buttons.

File: src/tests/Operate.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    expect(multiply).toBe('56');
+  });
+  test(' divide of 8 and 4 is equal to 2', () => {
+    const divide = operate(8, 4, '÷');
+    expect(divide).toBe('2');
+  });
+  test(' modulo of 9 and 4 is equal to 2', () => {
+    const divide = operate(9, 4, '%');
+    expect(divide).toBe('1');
+  });
```
Comment (afanimartin) [inline src/tests/Operate.test.js:23]: - Add a test case that throws an error when one tries to divide a number by zero.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @Mutai-Gilbert and @Yosefgeda, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Professional readme template used to document the project. Well done
- All test cases pass. Nice job

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Operate.test.js (hunk 2 above)
Comment (Mutai-Gilbert) [reply src/tests/Operate.test.js:23]: Done

File: src/tests/Calculate.test.js (hunk 1 above)
Comment (Mutai-Gilbert) [reply src/tests/Calculate.test.js]: Done

Comment (Shedrack-Sunday) [inline src/tests/Calculate.test.js]: - [x] You seem to have not made any commit from the last review request of the reviewer. 
- Kindly add missing tests for the `.` and `+/-` buttons.

File: src/tests/Operate.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    const multiply = operate(8, 7, 'x');
+    expect(multiply).toBe('56');
+  });
+  test(' divide of 8 and 4 is equal to 2', () => {
+    const divide = operate(8, 4, '÷');
+    expect(divide).toBe('2');
+  });
+  test(' modulo of 9 and 4 is equal to 2', () => {
+    const divide = operate(9, 4, '%');
+    expect(divide).toBe('1');
```
Comment (Shedrack-Sunday) [inline src/tests/Operate.test.js]: - [x]  Here also, You seem to have not made any commit from the last review request of the reviewer.
- Please add a test case that throws an error when one tries to divide a number by zero.

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @Yosefgeda        👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 


## Highlights:
✔️ good PR summary
✔️A few changes have been taken note of

  
# Required Changes ♻️

- [x] I must admit you have really put in the effort so far in this project, however, the changes required below need to be implemented :
- You didn't make the changes requested by the previous reviewer, as there is no commit from the last review 

_check the comment section for a detailed review_ 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/tests/Calculate.test.js (hunk 1 above)
Comment (Mutai-Gilbert) [reply src/tests/Calculate.test.js]: Done

File: src/tests/Operate.test.js (hunk 3 above)
Comment (Mutai-Gilbert) [reply src/tests/Operate.test.js]: Done

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
